the Lens.js

```diff
@@ -0,0 +1,215 @@
+	"translatorID": "b5f3b896-df57-4c68-8841-ea2171ee7419",
+	"label": "the Lens",
+	"creator": "Laurent Lhuillier",
+	"target": "^https?://www\\.lens\\.org",
+	"minVersion": "4.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2018-06-21 09:15:02"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	the Lens translator
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+	
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if(url.indexOf("search?") !== -1) {
+		return "multiple";
+	}
+		
+	if (url.indexOf("patent") !== -1 && getTitle(doc)) {
+		return "patent";
+	}
+}
+
+function getSearchResults(doc) {
+	return ZU.xpath(doc,'//*[@id="resultsTable"]/div[3]/child::*//*/h3/a');
+}
+
+function getTitle(doc) {
+	var title = ZU.xpathText(doc, '//h2[@class="doc-title"]/a');
+	if(title) {
+		if(title.toUpperCase() == title) {
+			title = ZU.capitalizeTitle(title, true);
+		}
+		return title.trim();
+	}
+}
+
+function scrape(doc) {
+	var newItem = new Zotero.Item("patent");
+	newItem.title = getTitle(doc);
+
+	var abstractNote = ZU.xpathText(doc, '//*[@id="patentFrontPage"]/div[1]/p');
+	newItem.abstractNote = abstractNote;
+
+	var patentNumber = ZU.xpathText(doc, '//*[@id="patentMain"]/div/header/div[2]/div/div[2]/a');
+	newItem.patentNumber = patentNumber;
+
+	var lensURL = ZU.xpathText(doc, '//*[@id="patentMain"]/div/header/div[2]/div/div[3]/a/@href');
+	newItem.url = 'https://www.lens.org' + lensURL;
+
+	var filingDate = ZU.xpathText(doc, '//*[@id="patentMain"]/div/header/div[1]/ul/li[1]/text()');
+	filingDate = new Date(filingDate);
+	newItem.filingDate = filingDate.toISOString().substr(0,10);
+
+	var ul = doc.getElementById("inventorFilter");
+	var liNodes = [];
+	for (var i = 0; i < ul.childNodes.length; i++) {
+		if (ul.childNodes[i].nodeName == "LI") {
+			liNodes.push(ul.childNodes[i]);
+		}
+	}
+	iAu = liNodes.length;
+
+	var authors = [];
+	for (iAu; iAu > 0; iAu--) {
+		authors[iAu] = ZU.xpathText(doc, '//*[@id="inventorFilter"]/li[' + iAu + ']/a');
+
+	newItem.creators.push(Zotero.Utilities.cleanAuthor(authors[iAu], "inventor", true));
+	}
+
+	var pdfurl = ZU.xpathText(doc, '//*[@id="patentFrontPage"]/div[2]/a/@href');
+	newItem.attachments.push({
+		title: 'Full Text PDF',
+		url: pdfurl,
+		mimeType: 'application/pdf'
+	});
```

Comment:
We should test here that `pdfurl` has some contents and only then do these lines here.